### ISSUE TITLE
invalid format character

### ISSUE THREAD

**Issue by jdalton** [omgmog/install-all-firefox#65]
When I run `install-all-firefox.sh` I get the error output but everything seems to finish fine.
```bash
install-all-firefox.sh: line 1001: printf: `.': invalid format character
```

Have you seen this?

**Comment by omgmog** [comment]
Well, line [`1001`](https://github.com/omgmog/install-all-firefox/blob/master/install-all-firefox.sh#L1001) is the `log` function, so something must be invoking it with a bad value. The only weird ones I can see are when we [have to redownload a dmg](https://github.com/omgmog/install-all-firefox/blob/master/install-all-firefox.sh#L790), as it contains a `.` followed by `\n`. And also the general log for [Installing a version](https://github.com/omgmog/install-all-firefox/blob/master/install-all-firefox.sh#L1090).

I'm not currently on my Mac so can't confirm that changing that would stop the error.

**Comment by jdalton** [comment]
Cool, I'll check it out.

**Comment by jdalton** [comment]
Modifying those spots to remove the `.\n` or the variable used doesn't prevent the error.

**Comment by omgmog** [comment]
Still poking around in the dark here, how about calling `printf` differently:

So instead of 

```bash
 printf "\n\033[32m$*\033[00m\n"
```

We can do

```bash
 printf "\n\033[32m%s\033[00m\n" $*
```

And if that works, we'll need to ensure that `printf` is being called correctly elsewhere too...

Cheers!

**Comment by jdalton** [comment]
That works with the side effect that all '\n' are now printed as literal "\n" but that's easy to fix.
I'll patch and create a PR.

**Comment by jdalton** [comment]
My python-fu is super weak, I'm a JS dev :P

**Comment by omgmog** [comment]
Cool, also this is shell! ;)

**Comment by jdalton** [comment]
Closed via #66.
